src/Key.php

```diff
@@ -5,18 +5,131 @@
 class Key
 {
     /** @var string */
-    public $id;
+    private $id;
 
     /** @var string */
-    public $secret;
+    private $secret;
+
+    /** @var string */
+    private $certificate;
+
+    /** @var string */
+    private $privateKey;
+
+    /** @var string */
+    private $type;
 
     /**
      * @param string $id
      * @param string $secret
      */
-    public function __construct($id, $secret)
+    public function __construct($id, $item)
     {
         $this->id = $id;
-        $this->secret = $secret;
+        $certificate = $this->getX509Certificate($item);
+        $privateKey = $this->getRSAPrivateKey($item);
+        if (($certificate || $privateKey)) {
+            $this->type = 'rsa';
+            if ($privateKey) {
+                $this->privateKey = $privateKey;
+            };
+            if ($certificate) {
+                $this->certificate = openssl_x509_read($certificate);
+            };
+            if ($certificate && $privateKey) {
+                // openssl_pkey_export($this->privateKey, $privateKey);
+            // openssl_x509_export($this->certificate, $certificate);
+            if (! openssl_x509_check_private_key(
+                $this->certificate, $this->privateKey)
+                ) {
```

Comment:
formatting is a bit off here - either inline or use consistent indenting and brackets. I've updated the php_cs_fixer on master to help with this
```php
function_name(
    $indented_arg1,
    $indented_arg2,
    $indented_arg3
);
```

Reply:
but generally inlining is easiest to read

Reply:
Fixed (Maybe?) in 665baabfcb62596b838b3c852613f45b81be700e